### PR TITLE
hack/install-etcd.sh: don't create unused temporary directories

### PR DESCRIPTION
**Pull request by php-coder**
This PR improves install-etcd.sh to not create temporary directories which this script doesn't use.

Also it improves error reporting for the case when temp directory is missing or user doesn't have permissions to access it:

```console
$ vagrant build-origin-base-images
Building base images...
/data/src/github.com/openshift/origin ~
WARNING: Current user doesn't have permissions to create directories in the /tmp/openshift
WARNING: Directory info:
WARNING: dr-xr-xr-x. 2 1000 1000 unconfined_u:object_r:user_tmp_t:s0 40 Apr  4 13:28 /tmp/openshift
WARNING: Current user info:
WARNING: uid=1000(vagrant) gid=1000(vagrant) groups=1000(vagrant),1001(docker),10001(rkt),427680(_subgroup) context=unconfined_u:unconfined_r:unconfined_t:s0-s0:c0.c1023
WARNING: SELinux state: Permissive
WARNING: audit logs:
<no matches>
WARNING: end of audit logs
mkdir: cannot create directory ‘/tmp/openshift/build-base-images’: Permission denied
mkdir: cannot create directory ‘/tmp/openshift/build-base-images’: Permission denied
mkdir: cannot create directory ‘/tmp/openshift/build-base-images’: Permission denied
mkdir: cannot create directory ‘/tmp/openshift/build-base-images’: Permission denied
[ERROR] PID 25705: hack/lib/util/environment.sh:157: `mkdir -p "${LOG_DIR}" "${VOLUME_DIR}" "${ARTIFACT_DIR}" "${FAKE_HOME_DIR}"` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/lib/util/environment.sh:157: `mkdir -p "${LOG_DIR}" "${VOLUME_DIR}" "${ARTIFACT_DIR}" "${FAKE_HOME_DIR}"`
[INFO] 		  2: hack/lib/init.sh:57: os::util::environment::setup_tmpdir_vars
[INFO] 		  3: hack/build-base-images.sh:8: source
[INFO]   Exiting with code 1.
```

PTAL @stevekuznetsov

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Only thing I would add would be audit search for the `$rootbasedir`

```
sudo ausearch --file "${rootbasedir}"
```

**Comment by php-coder** [comment]
> Only thing I would add would be audit search for the $rootbasedir

Thanks, I've added it.

File: hack/install-etcd.sh [hunk 1]
```diff
@@ -1,4 +1,9 @@
 #!/bin/bash
+
+# This prevents creation of temporary directories inside of the /tmp. This
+# script don't need them and creating them under root user could lead to other
+# scripts failing down the road.
+OS_TMP_ENV_SET=install-etcd
```
Comment (php-coder) [inline hack/install-etcd.sh]: @stevekuznetsov WDYT about this fix?

While I'm still testing it, the path that was causing a failure is already passing.

The benefits of it:
- it's short
- it makes script not create directories which this script isn't used
- we don't need to modify `vagrant install-origin` at all (it's Ruby code and AFIU it's not developing now)

File: hack/lib/cleanup.sh [hunk 2]
```diff
@@ -94,8 +94,8 @@ function os::cleanup::tmpdir() {
         done
     fi
 
-		# delete any sub directory underneath BASETMPDIR
-    for directory in $( find "${BASETMPDIR}" -d 2 ); do
+    # delete any sub directory underneath BASETMPDIR
+    for directory in $( find "${BASETMPDIR}" -mindepth 2 -maxdepth 2 ); do
```
Comment (php-coder) [inline hack/lib/cleanup.sh]: I'll remove this commit later, but I need to have this change to be sure that fork_ami job is passing.

**Comment by stevekuznetsov** [comment]
Just checked, no part of the `etcd` install needs `sudo` privileges. Correct fix here is to not run the command with `sudo` in the first place.

**Pull request closed without merging**